app/services/commitments/calculate_amount_service.rb

```diff
@@ -23,27 +23,10 @@ def call
     def commitment_amount_cents
       return 0 if !commitment || !invoice_subscription || commitment.amount_cents.zero?
 
-      # No proration when subscription is anniversary
-      return commitment.amount_cents if subscription.anniversary?
+      service = Commitments::HelperService.new(commitment:, invoice_subscription:, current_usage: false)
+      result = service.proration_coefficient
```

Comment:
I am not sure why invoice_subscription is instance variable in this class. If this class should calculate true-up based on invoice, I think that we should pass invoice reference to the constructor and make it an instance variable.
Btw, Can you please describe where this service will be called?